### DISCUSSION TITLE
Help for Yolo models/ 2d heatmaps

### DISCUSSION THREAD

**Discussion by Touville01** [roboflow/supervision#1216] [Q&A]
Hi everyone,

I'm currently working with the YOLOv8 models and have a few questions and issues that I hope you can help with.

1. Performance of YOLOv8x vs YOLOv8s:

Is YOLOv8x significantly more performant than YOLOv8s?
Does YOLOv8x absolutely require a CUDA GPU for reasonable performance? When I run YOLOv8x on my CPU, it takes considerably longer compared to YOLOv8s and YOLOv8n, and I'm not sure if this is expected behavior or if there's an issue with my setup.

2. Issues with YOLOv8s - "Time in Zone" Example:

I've been experiencing several errors with the YOLOv8s model, especially with the "time in zone" example and the provided sample video. The model struggles to accurately track the time some people spend in the designated zone. It detects a person, loses track of them after a couple of seconds, and then redetects them as a new person. This inconsistency is causing significant inaccuracies in tracking.

3. Creating a 2D Heatmap:

Additionally, I'd like to create a 2D heatmap to visualize the most frequented areas over the duration of the video. Does anyone have any ideas or suggestions on how to implement this?

Any help or insights you can provide would be greatly appreciated!

Thank you!

**Comment by SkalskiP** [comment]
Hi @Touville01 👋🏻 

If by "more performant" you mean better quality of detections, then yes. mAP@50-95 is one of the most popular benchmarks in computer vision, and YOLOv8s and YOLOv8x score 37.3 and 53.9 respectively on the COCO dataset. The difference might not be as significant when fine-tuning on a custom dataset, but there is still a substantial gap in detection quality between these models.

Regarding the performance difference between GPU and CPU, you can expect the models to be approximately 10 times slower on a CPU.

Problems with YOLOv8s are to be expected. In my opinion, YOLOv8s is practically unusable for such use cases. The quality of the predictions is simply too poor, which directly impacts the quality of tracking.

For heatmaps, I recommend using our [`HeatMapAnnotator`](https://supervision.roboflow.com/latest/detection/annotators/#__tabbed_1_18).
